types/types.proto

```diff
@@ -3,25 +3,24 @@ package types;
 
 import "github.com/gogo/protobuf/gogoproto/gogo.proto";
 
-
 // This file is copied from http://github.com/tendermint/abci
 
 //----------------------------------------
 // Request types
 
 message Request {
-  oneof value{
-    RequestEcho echo = 1;
-    RequestFlush flush = 2;
-    RequestInfo info = 3;
-    RequestSetOption set_option = 4;
-    RequestDeliverTx deliver_tx = 5;
-    RequestCheckTx check_tx = 6;
-    RequestCommit commit = 7;
-    RequestQuery query = 8;
-    RequestInitChain init_chain = 9;
-    RequestBeginBlock begin_block = 10;
+  oneof value {
```

Comment:
why this change?

Reply:
Just to screw with third party abci implementations?

Reply:
Oh - probably to correspond with the numbering for Response

Reply:
> Just to screw with third party abci implementations?

They should be using the protobuf file so hopefully they can just recompile.

Reply:
The order was changed to reflect the order of messages going into the app, more or less.

Reply:
after init_chain you can query, but doesn't make sense before then.  deliver is between begin and end block.  I think check_tx could go above begin_block, but it's nicer to keep those two side by side.  Commit comes at the very end.